src/ImageManipulation.php

```diff
@@ -1057,4 +1069,204 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    /**
+     * Create an equivalent DBFile object
+     * @return AssetContainer
+     */
+    private function copyImageBackend(): AssetContainer
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $backend = $this->getImageBackend();
+        if ($backend) {
+            $file->setImageBackend($backend);
+        }
+
+        return $file->setOriginal($this);
+    }
+
+    /**
+     * Add a HTML attribute when rendering this object. This method is immutable, it will return you a copy of the
+     * original object.
+     * @param string $name
+     * @param mixed $value
+     * @return AssetContainer
+     */
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->initAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    /**
+     * Initialise the attirbutes on this object. Should only be called imiditely after intialising the object.
+     * @param array $attributes
+     * @internal
+     */
+    public function initAttributes(array $attributes)
+    {
+        // This would normally be in a constructor, but there's no way for a trait to interact with that.
+        $this->attributes = $attributes;
+    }
+
+    /**
+     * Retrieve the value of an HTML attribute
+     * @param $name
+     * @return mixed|null
+     */
+    public function getAttribute($name)
+    {
+        $attributes = $this->getAttributes();
+
+        if (isset($attributes[$name])) {
+            return $attributes[$name];
+        }
+
+        return null;
+    }
+
+    /**
+     * Allows customization through an 'updateAttributes' hook on the base class.
+     * Existing attributes are passed in as the first argument and can be manipulated,
+     * but any attributes added through a subclass implementation won't be included.
+     *
+     * @return array
+     */
+    public function getAttributes()
+    {
+        $attributes = [];
+        if ($this->getIsImage()) {
+            $attributes = [
+                'width' => $this->getWidth(),
+                'height' => $this->getHeight(),
+                'alt' => $this->getTitle(),
+                'src' => $this->getURL()
+            ];
+
+            if ($this->IsLazyLoaded()) {
+                $attributes['loading'] = 'lazy';
+            }
+        }
+
+        $attributes = array_merge($attributes, $this->attributes);
+
+        if (isset($attributes['loading']) && $attributes['loading'] === 'eager') {
```

Comment:
This bit unset is here to make sure we don't have a `loading="eager"` snippet in our HTML